duneana/CAFMaker/CAFMaker_module.cc

```diff
@@ -490,19 +676,20 @@ namespace caf {
   {
     fMetaTree->Fill();
 
-    if(fOutFile){
-      fOutFile->cd();
-      fTree->Write();
-      fMetaTree->Write();
-      fOutFile->Close();
-    }
+    fOutFile->cd();
+    fTree->Write();
+    fMetaTree->Write();
+    fGENIETree->Write();
 
     if(fFlatFile){
       fFlatFile->cd();
       fFlatTree->Write();
-      fMetaTree->Write();
+      fMetaTree->CloneTree()->Write();
+      fGENIETree->CloneTree()->Write();
       fFlatFile->Close();
     }
+
+    fOutFile->Close();
```

Comment:
TFileSerivce management if possible

Reply:
After checking with ART people, TFileService cannot manage multiple output files. It is necessary to deal with any additional file manually if required. However, there would be the possibility to create subdirectories, but I would be afraid this makes this flat file kind of useless (which it might already be) if it is among other structured data in the same root file (for dictionary errors).

Reply:
A solution could be to see if there is any use for this flat version that might possibly be removed from the code.